File: data/data.code.lua.server/ReqTestPhp.lua
module("ReqTestPhp", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqTestPhp})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.u64 = 0
	self.strxx = 0
	self.msg_req = nil
	self.msg_opt = nil
	self.msg_rep = {}
end


function decode(self, pack)
	self.u64 = pack:readU64()
	self.strxx = pack:readString()
	self.msg_req = MsgTestPhp.new():decode(pack)
	 local msg_opt_flag = pack:readU8()
	if msg_opt_flag then
		self.msg_opt = MsgTestPhp.new():decode(pack)
	end
	local msg_rep_count = pack:readU16()
	for i = 1, msg_rep_count do
		table.insert(self.msg_rep, MsgTestPhp.new():decode(pack))
	end
	return self
end

File: data/data.code.golang.server/req_test_php.go
package proto

import (
	"packet"
)

type ReqTestPhp struct {
	U64                      uint64
	Strxx                    string
	MsgReq                   *MsgTestPhp
	MsgOpt                   *MsgTestPhp
	MsgRep                   []*MsgTestPhp
}

func ReqTestPhpDecode(pack *packet.Packet) *ReqTestPhp {
	reqTestPhp := &ReqTestPhp{}

	reqTestPhp.U64 = pack.ReadUint64()
	reqTestPhp.Strxx = pack.ReadString()
	reqTestPhp.MsgReq = MsgTestPhpDecode(pack)
	MsgOptFlag := pack.ReadUint8()
	if MsgOptFlag == 1 {
		reqTestPhp.MsgOpt = MsgTestPhpDecode(pack)
	}
	MsgRepCount := pack.ReadUint16()
	for ;MsgRepCount > 0; MsgRepCount-- {
		reqTestPhp.MsgRep = append(reqTestPhp.MsgRep, MsgTestPhpDecode(pack))
	}
	return reqTestPhp
}
